### PR TITLE
Add https-upgrade module and update to Kotlin 1.8.10

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1204775961153146/f

### Description
Updates to Kotlin 1.8.10.
- Reverts https://github.com/duckduckgo/Android/pull/3234
- With additional changes from https://github.com/duckduckgo/Android/pull/3361

### Steps to test this PR
- [x] Smoke test

### CODE REVIEW

**Review by aaronhook — COMMENTED** [review]
Very staight-foward. Had some trouble within

**Comment by malmstein** [comment]
@joshliebe  is this still valid? can we close?

File: httpsupgrade/httpsupgrade-api/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
```
Comment (joshliebe) [inline httpsupgrade/httpsupgrade-api/build.gradle:2]:
```suggestion
 * Copyright (c) 2023 DuckDuckGo
```

File: httpsupgrade/httpsupgrade-impl/build.gradle [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
```
Comment (joshliebe) [inline httpsupgrade/httpsupgrade-impl/build.gradle:2]:
```suggestion
 * Copyright (c) 2023 DuckDuckGo
```

File: httpsupgrade/httpsupgrade-store/build.gradle [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
```
Comment (joshliebe) [inline httpsupgrade/httpsupgrade-store/build.gradle:2]:
```suggestion
 * Copyright (c) 2023 DuckDuckGo
```

File: app/src/main/java/com/duckduckgo/app/di/JsonModule.kt [hunk 4]
```diff
@@ -32,6 +32,7 @@ object JsonModule {
     @SingleInstanceIn(AppScope::class)
     fun moshi(): Moshi = Moshi.Builder()
         .add(ActionJsonAdapter())
+        // FIXME we should not access HttpsFalsePositivesJsonAdapter directly here because it's in impl module
```
Comment (joshliebe) [inline app/src/main/java/com/duckduckgo/app/di/JsonModule.kt:35]: And this

File: httpsupgrade/httpsupgrade-impl/src/main/java/com/duckduckgo/httpsupgrade/impl/HttpsBloomFilterFactory.kt [hunk 5]
```diff
@@ -14,29 +14,26 @@
...
 
 interface HttpsBloomFilterFactory {
     fun create(): BloomFilter?
 }
 
-@ContributesBinding(AppScope::class)
-class HttpsBloomFilterFactoryImpl @Inject constructor(
-    private val dao: HttpsBloomFilterSpecDao,
+class HttpsBloomFilterFactoryImpl constructor(
+    private val dao: com.duckduckgo.httpsupgrade.store.HttpsBloomFilterSpecDao,
```
Comment (joshliebe) [inline httpsupgrade/httpsupgrade-impl/src/main/java/com/duckduckgo/httpsupgrade/impl/HttpsBloomFilterFactory.kt:36]: `com.duckduckgo.httpsupgrade.store.HttpsBloomFilterSpecDao` -> `HttpsBloomFilterSpecDao`

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! Great job :tada: :crossed_fingers: :shipit:

**Review by nalcalag — APPROVED** [review]
`common-ui` module looks good too. It is a formatting fix 💯

**Pull request merged**